0_download_data/correctionutils.py

```diff
@@ -74,41 +78,45 @@ def rename_images(path_to_images: pathlib.Path, output_folder_name: str):
         "01_3_4",
     )
 
+    # if any of the images contain the string (e.g. contains the channel number), then the images should be left alone
     for images in path_to_images.iterdir():
-        if any(x in str(images) for x in keywords):
+        if any(x in str(images) for x in channel_number_added):
             print("The metadata for this plate has already been corrected!")
             break
-        else:
-            if dapi_id in str(images):
-                DAPI_name = images.name.split("_")
-                well = DAPI_name[0]
-                site = DAPI_name[2]
-                channel_name = DAPI_name[3]
-                plate = DAPI_name[4].split(".")[0]
-
-                new_DAPI_name = f"{output_folder_name}/{well}_01_1_{site}_{channel_name}_{plate}.tif"
-                Path(images).rename(Path(new_DAPI_name))
-                print(f"DAPI {well}_{site} image has been renamed")
-
-            if gfp_id in str(images):
-                GFP_name = images.name.split("_")
-                well = GFP_name[0]
-                site = GFP_name[2]
-                channel_name = GFP_name[3]
-                plate = GFP_name[4].split(".")[0]
-
-                new_GFP_name = f"{output_folder_name}/{well}_01_2_{site}_{channel_name}_{plate}.tif"
-                Path(images).rename(Path(new_GFP_name))
-                print(f"GFP {well}_{site} image has been renamed")
-
-            if actin_id in str(images):
-                Actin_name = images.name.split("_")
-                well = Actin_name[0]
-                site = Actin_name[2]
-                # replace Actin with RFP to keep metadata consistent between datasets
-                channel_name = "RFP"
-                plate = Actin_name[4].split(".")[0]
-
-                new_Actin_name = f"{output_folder_name}/{well}_01_3_{site}_{channel_name}_{plate}.tif"
-                Path(images).rename(Path(new_Actin_name))
-                print(f"Actin (now RFP) {well}_{site} image has been renamed")
+
+        # this will correct the images with DAPI in the name and give the correct channel number
```

Comment:
There's quite a bit of duplication here, is there any way to make this DRYer?

I think you could do something like:

```python
# Define your channels in a dictionary instead of a string per variable
channel_id_dict = {
    "DAPI": {"id": "01_1"},
    "GFP": {"id": "01_2"},
    "Actin": {"id": "01_3", "channel_name": "RFP"}
}

for channel in channel_id_dict:
    channel_id = channel_id_dict[channel]["id"]
    if "channel_name" in channel_id_dict[channel].keys():
        channel_name = channel_id_dict[channel]["name"]
    else:
        channel_name = channel

    if channel in str(images):
        well, _, site, _, plate = images.name.split("_")
        plate = plate.split(".")[0]

        new_channel_name = f"{output_folder_name}/{well}_{channel_id}_{site}_{channel_name}_{plate}.tif"
        Path(images).rename(Path(new_channel_name))
```